eppoclient/bandits_test.go

```diff
@@ -0,0 +1,134 @@
+package eppoclient
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type banditTest struct {
+	Flag         string
+	DefaultValue string
+	Subjects     []struct {
+		SubjectKey        string
+		SubjectAttributes struct {
+			Numeric     map[string]float64 `json:"numeric_attributes"`
+			Categorical map[string]string  `json:"categorical_attributes"`
+		}
+		Actions []struct {
+			ActionKey             string
+			NumericAttributes     map[string]float64
+			CategoricalAttributes map[string]string
+		}
+		Assignment BanditResult
+	}
+}
+
+func Test_InferContextAttributes(t *testing.T) {
+	attributes := Attributes{
+		"string": "blah",
+		"int":    42,
+		"bool":   true,
+	}
+	contextAttributes := InferContextAttributes(attributes)
+
+	expected := ContextAttributes{
+		Numeric: map[string]float64{
+			"int": 42.0,
+		},
+		Categorical: map[string]string{
+			"string": "blah",
+			"bool":   "true",
+		},
+	}
+
+	assert.Equal(t, expected, contextAttributes)
+}
+
+func Test_bandits_sdkTestData(t *testing.T) {
+	flags := readJsonFile[ufcResponse]("test-data/ufc/bandit-flags-v1.json")
+	bandits := readJsonFile[banditResponse]("test-data/ufc/bandit-models-v1.json")
+	configStore := configurationStore{
+		flags:   flags.Flags,
+		bandits: bandits.Bandits,
+	}
+	logger := new(mockLogger)
+	logger.Mock.On("LogAssignment", mock.Anything).Return()
+	logger.Mock.On("LogBanditAction", mock.Anything).Return()
+	client := newEppoClient(&configStore, nil, nil, logger)
+
+	tests := readJsonDirectory[banditTest]("test-data/ufc/bandit-tests/")
+	for file, test := range tests {
+		t.Run(file, func(t *testing.T) {
+			for _, subject := range test.Subjects {
+				t.Run(subject.SubjectKey, func(t *testing.T) {
+					actions := make(map[string]ContextAttributes)
+					for _, a := range subject.Actions {
+						actions[a.ActionKey] = ContextAttributes{
+							Numeric:     a.NumericAttributes,
+							Categorical: a.CategoricalAttributes,
+						}
+					}
+
+					result := client.GetBanditAction(
+						test.Flag,
+						subject.SubjectKey,
+						ContextAttributes{
+							Numeric:     subject.SubjectAttributes.Numeric,
+							Categorical: subject.SubjectAttributes.Categorical,
+						},
+						actions,
+						test.DefaultValue)
+
+					assert.Equal(t, subject.Assignment, result)
+				})
+			}
+		})
+	}
+}
+
+func readJsonFile[T any](filePath string) T {
```

Comment:
This and other processing and serving test data files we already need to do for [`eppoclient_e2e_test.go`](https://github.com/Eppo-exp/golang-sdk/blob/main/eppoclient/eppoclient_e2e_test.go#L92), so consider consolidating/deduplicating the code into some sort of reusable helper function.